### PR TITLE
Fix: Update to the latest Cairo version

### LINKED ISSUE #7: Update to last cairo version
## Description 📹

**Update the project to the latest version of Cairo.**  
Ensure compatibility with all components and update dependencies accordingly.

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:
   
```bash
   git checkout -b fix-[issue-number]
```

2. **Implement Changes**:  

- [ ] Update the Cairo version in the project configuration.  
- [ ] Modify any breaking changes introduced by the new version of Cairo in the codebase.  
- [ ] Update relevant documentation to reflect the changes in the new Cairo version.  
- [ ] If any migrations or specific setup steps are required for the update, include them in the project setup documentation.  

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:
   
```bash
   git commit -m "Fix: Update to the latest Cairo version"
```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing Close #[issue_id].
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by Emmanex01**
Close #7

### CODE REVIEW

File: Scarb.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "resolver_delegation"
 version = "0.1.0"
 
 [dependencies]
-starknet = "2.3.1"
+starknet = ">=2.3.1"
```
Comment (Marchand-Nicolas) [inline Scarb.toml]: It doesn't build with scarb 2.8.5

Comment (Marchand-Nicolas) [reply Scarb.toml]: It's not a good idea to allow any higher version. If you don't succeed to make it build with the latest version (2.8.5), you can try one that is a bit less high (like 2.4 or 2.5, etc...)

And make sure it works by installing the corresponding scarb version: 
```
curl --proto '=https' --tlsv1.2 -sSf https://docs.swmansion.com/scarb/install.sh | sh -s -- -v <Version>
```

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]

File: Scarb.toml (hunk 1 above)
Comment (Emmanex01) [reply Scarb.toml]: Done.
I have updated it to the latest compatible version 2.4.0

**Comment by Emmanex01** [comment]
Done, cairo 2.6.3 updated

**Review by Marchand-Nicolas — APPROVED** [review]
Awesome, lgtm!

**Pull request merged**